docs/usage/CART_DISCOUNT_SYNC.md

```diff
@@ -67,28 +67,40 @@ Therefore, in order to resolve the actual ids of those references in the sync pr
 
 ````java
-// Build a CartDiscountQuery for fetching cart discounts from a source CTP project with all the needed references expanded for the sync
-final CartDiscountQuery cartDiscountQueryWithReferenceExpanded = CartDiscountReferenceResolutionUtils.buildCartDiscountQuery();
+// Build a CartDiscountQuery for fetching cart discounts from a source CTP project without any references expanded for the sync:
+final CartDiscountQuery cartDiscountQueryWithReferences = CartDiscountReferenceResolutionUtils.buildCartDiscountQuery();
 
 // Query all cart discounts (NOTE this is just for example, please adjust your logic)
 final List<CartDiscount> cartDiscounts =
     CtpQueryUtils
-        .queryAll(sphereClient, cartDiscountQueryWithReferenceExpanded, Function.identity())
+        .queryAll(sphereClient, cartDiscountQueryWithReferences, Function.identity())
         .thenApply(fetchedResources -> fetchedResources
             .stream()
             .flatMap(List::stream)
             .collect(Collectors.toList()))
         .toCompletableFuture()
         .join();
+````
+
+In order to transform and map the cartDiscount, 
+Initialize [`CartDiscountReferenceTransformService`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/cartdiscounts/service/CartDiscountReferenceTransformService.java) with `sphereClient` and cache(You can use your own cache implementation and pass the map).
+For cache implementation, you can refer an example class in the library - which implements the cache using caffeine library with an LRU based cache eviction strategy[`InMemoryReferenceIdToKeyCache`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/commons/utils/InMemoryReferenceIdToKeyCache.java).
+Then call the transformCartDiscountReferences method with the `cartDiscounts` parameter as shown below:
 
-// Mapping from CartDiscount to CartDiscountDraft with considering reference resolution.
-final List<CartDiscountDraft> cartDiscountDrafts = CartDiscountReferenceResolutionUtils.mapToCartDiscountDrafts(cartDiscounts);
+````java
+// Fetch(Id to key values for references) into the cache and map from CartDiscount to CartDiscountDraft using cache with considering reference resolution.
+final List<CartDiscountDraft> cartDiscountDrafts = CartDiscountReferenceTransformService.transformCartDiscountReferences(cartDiscounts);
 ````
 
+The cache here is used for a better performance. 
+Instead of expanding the references in the query for cartDiscount resource. `CartDiscountReferenceTransformService` will execute a query to fetch key values for the Ids(nonCached) and store in cache. These cached id to key values then can be used by another resource for resolving its references instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve the overall performance of the sync library and commercetools API.
```

Comment:
```suggestion
Instead of expanding the references in the query for cartDiscount resource. `CartDiscountReferenceTransformService` will execute a query to fetch key-to-id pairs and store in cache. These cached id to key values then can be used by another resource for resolving its references instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve the overall performance of the sync library and commercetools API.
```

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/710/commits/ab0e47f74ff70fe50190e2f11de65a9406caab83